internal/kubeapi/patch.go

```diff
@@ -117,7 +117,7 @@ func (patch *JSON6902) Replace(path ...string) func(value interface{}) *JSON6902
 func (patch JSON6902) Bytes() ([]byte, error) { return patch.Data(nil) }
 
 // Data returns the JSON representation of patch.
-func (patch JSON6902) Data(runtime.Object) ([]byte, error) { return json.Marshal(patch) }
+func (patch JSON6902) Data(client.Object) ([]byte, error) { return json.Marshal(patch) }
```

Comment:
We discussed this in a meeting, but in short, `r.patch` takes a `client.Patch` param, and that `client.Patch` interface has a `Data` method that takes a `client.Object` now rather than a `runtime.Object` -- though `client.Object` actually includes `runtime.Object`; ergo, changing `runtime.Object` => `client.Object` doesn't really require many other changes from us and is required for our JSON6902 type to satisfy the `client.Patch` interface

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Patch
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Object
https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#Object